admin/src/components/ReactIconsSelector/index.tsx

```diff
@@ -122,49 +138,93 @@ const ReactIconsSelector: React.FC<IReactIconsSelector> = ({
         required={required}
         error={error}
         startAction={
-          <FieldAction onClick={toggleModal}>
-            {value ? <IconComponent icon={value} /> : <ReactIcons.TbSearch />}
+          <FieldAction onClick={toggleModal} style={inputButtonStyle} left="0">
+            {value ? (
+              <IconComponent icon={value} />
+            ) : (
+              <ReactIcons.TbSearch width="1.25rem" height="1.25rem" />
+            )}
           </FieldAction>
         }
         endAction={
           !!value && (
-            <FieldAction onClick={() => changeIcon("")}>
+            <FieldAction
+              onClick={() => changeIcon("")}
+              style={inputButtonStyle}
+              right="0"
+            >
               <ReactIcons.TbX />
             </FieldAction>
           )
         }
       />
 
       {isModalVisible && (
-        <ModalLayout onClose={toggleModal} labelledBy="title">
+        <ModalLayout
+          onClose={toggleModal}
+          labelledBy="title"
+          width="65vw !important"
+        >
           <ModalHeader>
             <Typography fontWeight="bold" id="title">
               Select icon
             </Typography>
           </ModalHeader>
-          <ModalBody>
-            <Flex direction="row" wrap="wrap" gap={2}>
-              {selectedIconLibrary ? (
-                selectableIcons.length > 0 ? (
-                  selectableIcons.map((icon) => (
-                    <Box
-                      key={icon}
-                      variant="secondary"
-                      onClick={() => {
-                        toggleModal();
-                        changeIcon(icon);
-                      }}
-                    >
-                      <IconComponent size={30} icon={icon} />
-                    </Box>
-                  ))
-                ) : (
-                  <Typography variant="pi">No icons available</Typography>
-                )
-              ) : (
-                <Typography variant="pi">No icon library selected</Typography>
-              )}
-            </Flex>
+          <ModalBody
+            minHeight="70vh"
+            maxHeight="unset !important"
+            overflow="hidden"
+          >
+            <TextInput
+              id="icon-search"
+              name="icon-search"
+              label={intlLabel && formatMessage(intlLabel)}
+              value={searchTerm}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setSearchTerm(e.target.value)
+              }
+              placeholder="Search icons..."
+              margin={"15px"}
+            />
+            <Grid
```

Comment:
The Grid layout has a problem on smaller screens, because you get a vertical scroll bar with it. Thats why I used Flex with wrap. Is there any advantage from using Grid instead of Flex?

Reply:
@DanielPantle I don't know if you caught it, but I wasn't using the strapi design system grid, I was using the react-window library. Fixed and styled scrollbar so far.